seisbench/models/dpp.py

```diff
@@ -223,3 +526,525 @@ def forward(self, x):
+                )
+            elif start_element is None and end_element is not None:
+                to_write.append(end_element[2])
+            else:
+                # both start_element and end_element are active
+                if end_element[0] < start_element[0] or (
+                    strict and end_element[0] == start_element[0]
+                ):
+                    to_write.append(end_element[2])
+                    start_sorted.put(start_element)
+                else:
+                    active.put(
+                        (
+                            start_element[2].stats.endtime,
+                            start_element[1],
+                            start_element[2],
+                        )
+                    )
+                    active.put(end_element)
+
+            if not strict and active.qsize() == 0 and len(to_write) != 0:
+                t0 = min(trace.stats.starttime for trace in to_write)
+                t1 = max(trace.stats.endtime for trace in to_write)
+
+                data = np.zeros(
+                    (len(component_order), int((t1 - t0) * sampling_rate + 2))
+                )  # +2 avoids fractional errors
+
+                for trace in to_write:
+                    p = int((trace.stats.starttime - t0) * sampling_rate)
+                    cidx = comp_dict[trace.id[-1]]
+                    data[cidx, p : p + len(trace.data)] = trace.data
+
+                data = data[:, :-1]  # Remove fractional error +1
+
+                output_times.append(t0)
+                output_data.append(data)
+
+                to_write = []
+
+            if strict and active.qsize() == len(component_order):
+                traces = []
+                while not active.empty():
+                    traces.append(active.get()[2])
+
+                t0 = max(trace.stats.starttime for trace in traces)
+                t1 = min(trace.stats.endtime for trace in traces)
+
+                short_traces = [trace.slice(t0, t1) for trace in traces]
+                data = np.zeros(
+                    (len(component_order), len(short_traces[0].data) + 2)
+                )  # +2 avoids fractional errors
+                for trace in short_traces:
+                    cidx = comp_dict[trace.id[-1]]
+                    data[cidx, : len(trace.data)] = trace.data
+
+                data = data[:, :-2]  # Remove fractional error +2
+
+                output_times.append(t0)
+                output_data.append(data)
+
+                for trace in traces:
+                    if t1 < trace.stats.endtime:
+                        start_sorted.put((t1, seqnum, trace.slice(starttime=t1)))
+                        seqnum += 1
+
+        return output_times, output_data
+
+    def classify_aggregate(self, annotations, picks_in, argdict):
+        """
+        Converts the annotations to discrete picks using :py:func:`~seisbench.models.base.WaveformModel.picks_from_annotations`.
```

Comment:
Docstring has been adapted, now it looks as shown below.
The `[phase]_threshold` key is still used, but with slightly different meaning that in detection models, since here it indicates the threshold (by default 0.5) used in the overwritten `picks_from_annotations` function to identify the pick onset from the MCD samples.

```
def classify_aggregate(self, annotations, argdict):                         
    """                                                                     
    Converts the annotations (MCD samples) to discrete picks using :py:func:`picks_from_annotations`.
    Thresholds for pick onsets (transition from noise to signal) are derived from the argdict at keys "[phase]_threshold".
                                                                            
    :param annotations: See description in superclass                       
    :param argdict: See description in superclass                           
    :return: List of picks                                                  
    """       
```